homeassistant/components/browan/sensor.py

```diff
@@ -0,0 +1,152 @@
+
+
+async def _async_setup_entity(
+    hass: HomeAssistant,
+    async_add_entities: AddEntitiesCallback,
+    config_entry: ConfigEntry,
+) -> None:
+    """Set up Browan sensor."""
+    entities = []
+
+    # Device should not generate any error as it is returned by
+    # the config flow selector, but catch the error in case something
+    # goes wrong somewhere
+    if not config_entry.data["model"].isalnum():
+        _LOGGER.error(
+            'Device name "%s" from %s is invalid',
+            config_entry.data["model"],
+            MANUFACTURER.capitalize(),
+        )
+        return
+    try:
+        device = getattr(devices, f'Hass{config_entry.data["model"]}')
+    except AttributeError:
+        _LOGGER.error(
+            'Device "%s" from %s is unknown',
+            config_entry.data["model"],
+            MANUFACTURER.capitalize(),
+        )
+        return
+
+    coordinator = LorawanSensorCoordinator(hass, config_entry, device.parse_uplink)
+    for sensor in device.supported_sensors():
+        entities.append(LorawanSensorEntity(hass, config_entry, coordinator, sensor))
+    async_add_entities(entities)
+    await coordinator.subscribe()
+
+
+class LorawanSensorCoordinator(DataUpdateCoordinator):
+    """Allows to update all entities with one payload reception."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        uplink_parser: Callable[[Uplink], Awaitable[Uplink]],
+    ) -> None:
+        """Initialize the coordinator."""
+        self._config = config_entry
+        self._uplink_parser = uplink_parser
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"LorawanSensorCoordinator.{config_entry.title}",
+        )
+
+    async def _message_received(self, msg: ReceiveMessage) -> None:
+        """Handle uplink, parse and normalize it."""
+        uplink = json.loads(msg.payload)
+        uplink = TTN.normalize_uplink(uplink)
+        await self._uplink_parser(uplink)
+
+        self.async_set_updated_data(uplink)
+
+    async def subscribe(self) -> None:
+        """Subscribe to MQTT messages and handle them."""
+        topic = TTN_TOPIC.replace("<DEVICE_ID>", self._config.title)
+        await self.hass.components.mqtt.async_subscribe(topic, self._message_received)
+
+
+class LorawanSensorEntity(CoordinatorEntity, SensorEntity):
+    """Representation of a sensor entity that can be updated using LoRaWAN."""
+
+    _attr_has_entity_name = True
+    _attr_state_class = SensorStateClass.MEASUREMENT
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        coordinator: LorawanSensorCoordinator,
+        sensor: type[SensorTypes.SensorType],
+    ) -> None:
+        """Initialize the sensor."""
+        self._attr_unique_id = f"{config_entry.unique_id}_{sensor.NAME}"
+        self._attr_device_class = sensor.DEVICE_CLASS
+        self._attr_name = sensor.NAME
```

Comment:
set `_attr_has_entity_name=True` instead and put your names in strings.json (https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations)

Reply:
Thank you for the review and the `EntityDescription` hint, I was not aware of it.

Regarding `_attr_has_entity_name` it is set as a class attribute, is it ok?

https://github.com/home-assistant/core/blob/681b6e1886eaa76b5b668e6e46e6d23390527481/homeassistant/components/browan/sensor.py#L156

Then I also fixed the names in strings.json/translations thanks to the `EntityDescription`  in c734bf0025